lane.py

```diff
@@ -107,28 +107,33 @@ class FacetsWithEntryPoint(FacetConstants):
     """Basic Facets class that knows how to filter a query based on a
     selected EntryPoint.
     """
-    def __init__(self, entrypoint=None, **kwargs):
+    def __init__(self, entrypoint=None, entrypoint_is_default=False, **kwargs):
         """Constructor.
 
         :param entrypoint: An EntryPoint (optional).
+        :param entrypoint_is_default: If this is True, then `entrypoint`
+            is a default value and was not determined by a user's
+            explicit choice.
         :param kwargs: Other arguments may be supplied based on user
             input, but the default implementation is to ignore them.
         """
         self.entrypoint = entrypoint
+        self.entrypoint_is_default = entrypoint_is_default
         self.constructor_kwargs = kwargs
 
     def navigate(self, entrypoint):
         """Create a very similar FacetsWithEntryPoint that points to
         a different EntryPoint.
         """
         return self.__class__(
-            entrypoint=entrypoint, **self.constructor_kwargs
+            entrypoint=entrypoint, entrypoint_is_default=False,
```

Comment:
When you call navigate() you are making some kind of choice, so the default-ness is gone.